plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryClient.java

```diff
@@ -292,16 +288,12 @@ String selectSql(TableId table, String formattedColumns)
         return format("SELECT %s FROM `%s`", formattedColumns, tableName);
     }
 
-    private String fullTableName(TableId tableId)
+    private String fullTableName(TableId remoteTableId)
     {
-        String remoteSchemaName = toRemoteDataset(tableId.getProject(), tableId.getDataset())
-                .map(RemoteDatabaseObject::getOnlyRemoteName)
-                .orElse(tableId.getDataset());
-        String remoteTableName = toRemoteTable(tableId.getProject(), remoteSchemaName, tableId.getTable())
-                .map(RemoteDatabaseObject::getOnlyRemoteName)
-                .orElse(tableId.getTable());
-        tableId = TableId.of(tableId.getProject(), remoteSchemaName, remoteTableName);
-        return format("%s.%s.%s", tableId.getProject(), tableId.getDataset(), tableId.getTable());
```

Comment:
This was the issue - correct? We already had the remote name but were trying to translate it anyway.

Reply:
Correct, here and the above places were the issue.